app/src/main/java/com/github/polypoly/app/network/IRemoteStorage.kt

```diff
@@ -1,7 +1,9 @@
 package com.github.polypoly.app.network
 
+import com.github.polypoly.app.game.GameLobby
 import com.github.polypoly.app.game.User
 import java.util.concurrent.CompletableFuture
+import java.util.concurrent.Future
 
 interface IRemoteStorage {
```

Comment:
Your remote storage is poorly abstracted. It should not be directly responsible for managing GameLobbies